### PR TITLE
feat(filecoin-proofs): add APIs for retrieving sealing and sector update keys

### LINKED ISSUE #1666: Method to generate sector key
There seems to be no good way to generate a sector key today - which is needed for snapdeals unsealing, and possibly for future unsealing service-type things.

One could call Precommit1 and Precommit2 with null bytes (PC1 alone doesn't generate the sector key - https://github.com/filecoin-project/rust-fil-proofs/issues/1665)

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
This PR adds:
- `SealPreCommitPhase1Output::sealing_key_path` - returns path to "sealing key" (i.e. last layer)
- `SealPreCommitPhase1Output::read_sealing_key_range` - returns a range of nodes of the sealing key
- `filecoin_proofs::api::update::read_sector_key_range` - returns a range of nodes of the "sector update key" (i.e. the originally sealed sector, also called "old replica" within the context of sector updating)

Closes #1666
(and relates to #1667)

### CODE REVIEW

**Comment by DrPeterVanNostrand** [comment]
Hi @magik6k, if you have a chance, could you offer any feedback on this key retrieval API from a Lotus perspective?

Is retrieving a range of nodes from the sector update key sufficient, or would a different API be better for Lotus? I remember that you mentioned using file descriptors to retrieve keys, however that would require quite a bit of API refactoring.

File: filecoin-proofs/src/api/update.rs [hunk 1]
```diff
@@ -281,6 +282,28 @@ pub fn remove_encoded_data<Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>
...
+pub fn read_sector_key_range<Tree: MerkleTreeTrait>(
+    replica_old_path: &Path,
+    node_range: std::ops::Range<usize>,
+) -> Result<Vec<<Tree::Hasher as Hasher>::Domain>> {
+    let replica_old_file = OpenOptions::new()
+        .read(true)
+        .open(replica_old_path)
+        .with_context(|| format!("could not open path={:?}", replica_old_path))?;
+    let replica_old = unsafe {
+        MmapOptions::new()
```
Comment (cryptonemo) [inline filecoin-proofs/src/api/update.rs:295]: Might make sense to mmap just the range that's required here

**Review by magik6k — COMMENTED** [review]
I don't think the key-range-reading methods are useful by themselves to lotus / other proofs api consumers. For #1667 I'd imagine the proofs api would just have methods taking os pipes with (fr32-aligned) streams of sealed data + sector key data, and would get a decoded stream of unsealed data.

`sealing_key_path` Looks good, and should make fixing snapdeals unsealing much less hacky.

**Comment by DrPeterVanNostrand** [comment]
Replaced by PR #1687

**Pull request closed without merging**